app/controllers/automated_tests_controller.rb

```diff
@@ -1,15 +1,14 @@
 class AutomatedTestsController < ApplicationController
   include AutomatedTestsHelper
 
-  before_action      :authorize_only_for_admin,
-                     except: [:student_interface,
-                              :get_test_runs_students,
-                              :execute_test_run]
-
-  before_action      :authorize_for_student,
-                     only: [:student_interface,
-                            :get_test_runs_students,
-                            :execute_test_run]
+  before_action do
+    if params[:id].nil?
```

Comment:
It looks like none of the policies in `AutomatedTestPolicy` require a target (they're all based on the current user). This means you should be able to simplify this code by just doing an `authorize!` call, like in other files.

Reply:
Yes, the policies do not require a target but there is no specific model and table for automated tests. We should pass a target explicitly. I tried without the target like **"authorize!, with: :AutomatedTestPolicy"** but it is not looking up the policy correctly.